### ISSUE TITLE
Feature request: server side transitionTo hook.

### ISSUE THREAD

**Issue by BerkeleyTrue** [rackt/react-router#1566]
At the moment, transitionTo Hook only works client side. I have a use case where an endpoint with a param needs to detect if the param is valid, if not it need to redirect to another url. transitionTo works great for this, but server side react-router throws.

It be awesome if we could add a hook for transitionTo to happen server-side. In this hook it would be possible to do things like do a 302 redirect. This hook could be added as a prop to Router component and run func. 

The transitionTo method would not call external transitionTo hook if it detects its on the server instead of throwing and throw if it does not exist.

**Comment by maximilianschmitt** [comment]
Hi, I just found your post because I have the same problem: https://github.com/rackt/react-router/issues/1593

What are you currently doing to work around this limitation? I came up with this little piece of code, maybe it's helpful to you: https://gist.github.com/maximilianschmitt/3a14dffd6645dd4529f6

I'd love to hear about your approach in case you found a solution to your problem. Thanks!

**Comment by BerkeleyTrue** [comment]
I haven't implemented anything yet. Trying to understand what a PR to fix this would look like. 

Your fix seems interesting but hard to follow.

**Comment by goldensunliu** [comment]
:+1:

**Comment by Teaflavored** [comment]
I'm interested in this too, my work around right now is to use transitionTo hooks only in ComponentDidMount, which is called only on client side. I would love a clean way to do redirects on the server. The context I'm using this in is on my login in page. 

Here's my current page if anyone's interested, I have an authentication store keeping track of the user.

```
var React = require("react");
var Router = require("react-router");
var Navigation = Router.Navigation;
var fluxibleAddons = require("fluxible-addons-react");

//stores
var authenticationStore = require("../../stores/authentication_store");

//actions
var authenticate = require("../../actions/authenticate");

var Login = React.createClass({
    mixins: [Navigation],
    contextTypes: {
        getStore: React.PropTypes.func.isRequired,
        executeAction: React.PropTypes.func.isRequired
    },
    getInitialState: function () {
        return {
            email: "",
            password: ""
        };
    },
    getDefaultProps: function () {
        return {
            isLoggedIn: false
        }
    },
    componentDidMount: function () {
        if (this.props.isLoggedIn) {
            this.transitionTo("home");
        }
    },
    handleEmailchange: function (event) {
        this.setState({
            email: event.target.value
        });
    },
    handlePasswordChange : function () {
        this.setState({
            password: event.target.value
        });
    },
    handleLogin: function () {
        this.context.executeAction(authenticate, {
            body : this.state,
            component: this
        })
    },
    render: function () {
        var error = this.props.error;
        var errorNode;
        if (error) {
            errorNode = <div className="alert-danger alert">{error.message}</div>;
        } else {
            errorNode = "";
        }
        return (
            <div className="col-md-4 col-md-offset-4">
                <form action="javascript:void(0);" className="card">
                    {errorNode}
                    <div className="form-group">
                        <label htmlFor="username">E-mail</label>
                        <input type="text" className="form-control" onChange={this.handleEmailchange} />
                    </div>
                    <div className="form-group">
                        <label htmlFor="password">Password</label>
                        <input type="password" className="form-control" onChange={this.handlePasswordChange} />
                    </div>
                    <div className="form-group">
                        <input type="submit" value="Log In" className="btn btn-primary btn-block" onClick={this.handleLogin}/>
                    </div>
                </form>
            </div>
        );
    }
});

Login = fluxibleAddons.connectToStores(Login, [authenticationStore], function (context, props){
    return {
        error : context.getStore(authenticationStore).getError(),
        isLoggedIn : context.getStore(authenticationStore).isLoggedIn()
    };
});
module.exports = Login;
```

**Comment by djbobbydrake** [comment]
Am interested too. Right now there's a noticeable delay from when the page loads to when the redirect happens because the first page render is done server side.

**Comment by ryanflorence** [comment]
transitionTo works on the 0.13 branch, 1.0 branch server rendering is still baking, but transitioning will be supported still.

Pretty old API (0.12 I think) but it's always worked https://github.com/rackt/react-router-mega-demo/blob/master/app/handlers/CreateContact.js#L12

**Comment by ryanflorence** [comment]
I guess its possible that 0.13 doesn't allow this (probably an accident), but the 1.0 api will absolutely let you redirect via `transitionTo`

**Comment by Teaflavored** [comment]
The problem I'm having with this is that willTransitionTo is a static
member, and sometimes I want the context in order to grab the store to
check for data (such as authentication) to decide whether to transition or
not.
